run.sh

```diff
@@ -110,53 +113,22 @@ workload_1_golang() {
 }
 
 workload_7() {
-  local UPLOAD_OBJECT_PREFIX="test-object-$(date +'%s%N')-${API}-${OBJECT_SIZE}"
-  local UPLOAD_RESULTS_FILE="${WORKLOAD}_${API}_${OBJECT_SIZE}_UPLOAD"
-  local UPLOAD_RESULTS_FILE_PROCESSED="${UPLOAD_RESULTS_FILE}_PROCESSED"
-  /usr/bin/aggregate_upload_throughput_benchmark \
-    --labels=workload_7_upload \
-    --api="${API}" \
-    --object-count=1 \
-    --minimum-object-size="${OBJECT_SIZE}" \
-    --maximum-object-size="${OBJECT_SIZE}" \
-    --bucket-name="${BUCKET_NAME}" \
-    --object-prefix="${UPLOAD_OBJECT_PREFIX}" \
-    --iteration-count="${SAMPLES}" > "${UPLOAD_RESULTS_FILE}"
-  cat "${UPLOAD_RESULTS_FILE}" | \
-      grep -v "#.*" > "${UPLOAD_RESULTS_FILE_PROCESSED}"
-  tail -n +2 "${UPLOAD_RESULTS_FILE_PROCESSED}" | awk -F, '{print "throughput{workload="$2",Iteration="$1",ObjectCount="$3",ResumableUploadChunkSize="$4",ThreadCount="$5",Api="$6",GrpcChannelCount="$7",GrpcPluginConfig="$8",RestHttpVersion="$9",ClientPerThread="$10",StatusCode="$11",Peer="$12",BytesUploaded="$13",ElapsedMicroseconds="$14",IterationBytes="$15",IterationElapsedMicroseconds="$16",IterationCpuMicroseconds="$17"} "(($13/1024/1024)/($14/1000000))}'
-  rm "${UPLOAD_RESULTS_FILE_PROCESSED}" "${UPLOAD_RESULTS_FILE}"
-
-  local DOWNLOAD_OBJECT_PREFIX="test-object-$(date +'%s%N')-${API}-${OBJECT_SIZE}"
-  local DOWNLOAD_RESULTS_FILE="${WORKLOAD}_${API}_${OBJECT_SIZE}_DOWNLOAD"
-  local DOWNLOAD_RESULTS_FILE_PROCESSED="${DOWNLOAD_RESULTS_FILE}_PROCESSED"
-  /usr/bin/create_dataset \
-    --bucket-name="${BUCKET_NAME}" \
-    --object-prefix="${DOWNLOAD_OBJECT_PREFIX}" \
+  /usr/bin/storage_throughput_vs_cpu_benchmark \
     --minimum-object-size="${OBJECT_SIZE}" \
     --maximum-object-size="${OBJECT_SIZE}" \
-    --object-count=1 \
-    --thread-count=1 > /dev/null
-  /usr/bin/aggregate_download_throughput_benchmark \
-    --labels=workload_7_download \
-    --api="${API}" \
-    --bucket-name="${BUCKET_NAME}" \
-    --object-prefix="${DOWNLOAD_OBJECT_PREFIX}" \
-    --iteration-count="${SAMPLES}" > "${DOWNLOAD_RESULTS_FILE}"
-  cat "${DOWNLOAD_RESULTS_FILE}" | \
-      grep -v "#.*" > "${DOWNLOAD_RESULTS_FILE_PROCESSED}"
-  tail -n +2 "${DOWNLOAD_RESULTS_FILE_PROCESSED}" | awk -F, '{print "throughput{workload="$1",Iteration="$2",ObjectCount="$3",DatasetSize="$4",ThreadCount="$5",RepeatsPerIteration="$6",ReadSize="$7",ReadBufferSize="$8",Api="$9",GrpcChannelCount="$10",GrpcPluginConfig="$11",ClientPerThread="$12",StatusCode="$13",Peer="$14",BytesDownloaded="$15",ElapsedMicroseconds="$16",IterationBytes="$17",IterationElapsedMicroseconds="$18",IterationCpuMicroseconds="$19"} "(($4/1024/1024)/($16/1000000))}'
-  rm "${DOWNLOAD_RESULTS_FILE_PROCESSED}" "${DOWNLOAD_RESULTS_FILE}"
+    --region="${REGION}" \
+    --project-id="${PROJECT}" \
+    --enabled-transports="${API}" \
+    --minimum-sample-count="${SAMPLES}" \
+    --maximum-sample-count="${SAMPLES}" \
+    --upload-functions="${UPLOAD_FUNCTION}" | \
+    grep  -v "^#" | \
```

Comment:
FWIW, awk can skip lines matching (or not matching) a pattern:

https://unix.stackexchange.com/questions/286718/completely-ignore-lines-that-start-with-a-specific-pattern